src/allmydata/storage/http_server.py

```diff
@@ -542,10 +545,11 @@ def _read_encoded(
             raise _HTTPError(http.UNSUPPORTED_MEDIA_TYPE)
 
         # Make sure it's not too large:
-        request.content.seek(SEEK_END, 0)
-        if request.content.tell() > max_size:
+        request.content.seek(0, SEEK_END)
+        size = request.content.tell()
+        if size > max_size:
             raise _HTTPError(http.REQUEST_ENTITY_TOO_LARGE)
-        request.content.seek(SEEK_SET, 0)
+        request.content.seek(0, SEEK_SET)
```

Comment:
Interesting!

I guess this "worked" because `SEEK_SET == 0` and the mistake above with `SEEK_END` just silently made all requests look like they were 2 bytes long?

Does that mean the `max_size` check is untested?

Reply:
It's tested, just... the two code paths are identical in behavior. So a bug where codepath A is never called at all and only codepath B is called is hard to catch with black box tests.

So... how would you feel about doing some coverage mark testing, ala https://docs.rs/cov-mark/latest/cov_mark/?